src/fu-security-attr.c

```diff
@@ -203,6 +203,598 @@ fu_security_attr_get_name(FwupdSecurityAttr *attr)
+			g_ptr_array_add(strs, _("The system firmware is locked and not writable."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware is unlocked and writable by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_BLE) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			/* TRANSLATORS: longer success message (unlikely to see) */
+			g_ptr_array_add(strs, _("The system firmware cannot be easily unlocked."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware can be unlocked and written by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_SMM_BWP) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			/* TRANSLATORS: longer success message (unlikely to see) */
+			g_ptr_array_add(strs, _("The system firmware cannot be unlocked."));
+		} else {
+			g_ptr_array_add(
+			    strs,
+			    /* TRANSLATORS: longer failure message */
+			    _("The system firmware could be unlocked and written by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_SPI_DESCRIPTOR) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("Critical parts of the system firmware are protected."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("Critical parts of the system firmware could be changed "
+					  "by any user."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_PREBOOT_DMA_PROTECTION) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("System memory is protected from malicious devices "
+					  "before starting the OS."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("System memory is not protected from malicious devices "
+					  "before starting the OS."));
+			contact_hw_vendor = TRUE;
+		}
+	}
+	if (g_strcmp0(appstream_id, FWUPD_SECURITY_ATTR_ID_INTEL_BOOTGUARD_ENABLED) == 0) {
+		if (fwupd_security_attr_has_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS)) {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer success message (unlikely to see) */
+					_("System firmware is being checked by the processor at "
+					  "system start."));
+		} else {
+			g_ptr_array_add(strs,
+					/* TRANSLATORS: longer failure message */
+					_("System firmware is not being checked by the processor "
+					  "at system start."));
+			contact_hw_vendor = TRUE;
```

Comment:
This needs `check_bios_settings` too.

Reply:
I've heard of a vendor doing exactly that in the field -- but that was part of a BIOS update (i.e. `contact_hw_vendor`), not adding BIOS setting.

Reply:
Ah I see- makes sense.